### PR TITLE
All elections totals table

### LINKED ISSUE #5131: All elections totals table: Front-end to begin implementation of table
### Summary

**What we're after:**
All elections totals table: Front-end to begin implementation of table
Screen Shot 2022-03-23 at 1 46 04 PM

### Related issues
Epic: https://github.com/fecgov/fec-epics/issues/213

### Completion criteria

The All elections totals table is built

### Tech steps or considerations (optional)

- [x] What endpoint do we use for this?
  - [x] Use  new aggregate endpoints being created in this issue https://github.com/fecgov/openFEC/issues/5026
- [x] Fix wrapping disclaimer text on [contributions across time](https://github.com/fecgov/fec-cms/pull/5166#issuecomment-1115160985)

### PR DESCRIPTION
**Pull request by patphongs**
## Summary

- Resolves #5131 

All elections totals datatable using the new `/candidates/totals/aggregates/` endpoint.

### Required reviewers

1 front-end,  1 UX

## Impacted areas of the application

General components of the application that this PR will affect:

-  House and Senate elections overview pages

## Screenshots

Screen Shot 2022-05-16 at 12 30 44 PM
Screen Shot 2022-05-16 at 12 30 28 PM

## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
feature/5034-contributions-across-time | [link](https://github.com/fecgov/fec-cms/pull/5166)

## How to test

- checkout this branch
- `npm run build
- `cd fec && ./manage.py runserver`
- Go to the elections overview pages:
   - [Senate all elections totals table](http://localhost:8000/data/elections/senate/#totals-for-all-elections)
   - [House all elections totals table](http://localhost:8000/data/elections/house/#totals-for-all-elections)

### CODE REVIEW

**Review by cnlucas — APPROVED** [review]
Thanks for pairing!

**Comment by JonellaCulmer** [comment]
@patphongs Just a couple comments:
- There is a bold blue line missing from the top of the table. It should run along the length of feature and sit just above the column headers. 
Screen Shot 2022-05-13 at 10 49 14 AM

- The table looks different from the contributions across time feature. It's not contained within the gray background and the description "Compare the money..." is in bold.
- Also, should we edit the text to say "Compares..."? That's my typo - sorry I didn't see that sooner.

File: fec/data/templates/partials/house-senate-overview/election-data.jinja [hunk 1]
```diff
@@ -152,30 +155,42 @@
...
          <h2 class="heading__left">Totals for all {{ office | title }} elections</h2>
          <p class="t-sans t-bold t-low-height">Compare the money raised, spent, cash on hand and debts by {{ office | title }} election</p>
-         {{ select.two_year_select(cycles, cycle, id="cycle-4") }}
-         <div class="content__section--ruled-responsive t-serif">
-            <div class="heading--with-action t-small u-negative--top--margin t-serif">
-               <i class="data-disclaimer">Some kind of disclaimer text that clarifies something high-level that users need to know right away to put the data in context.</i>
-               <div class="heading__right">
-                  <a class="button button--alt js-ga-event" data-a11y-dialog-show="totals-for-all-elections-modal" data-ga-event="Totals for all elections methodology modal clicked" aria-controls="totals-for-all-elections-modal">Methodology</a>
+         {{ select.two_year_select(cycles, cycle, id="all-elections-totals") }}
+         <table id="election-totals-results"
```
Comment (johnnyporkchops) [inline fec/data/templates/partials/house-senate-overview/election-data.jinja:163]: @patphongs You could add `simple-table u-no-border` to table class  to get the top border that Jonella mentioned (`u-no-border` removes a double bottom border)

**Review by johnnyporkchops — CHANGES_REQUESTED** [review]
Works/Looks great, all the sorting and responsive behavior is working perfectly! Just a few html edits to for display.

File: fec/data/templates/partials/house-senate-overview/election-data.jinja [hunk 2]
```diff
@@ -87,45 +87,48 @@
...
+                        </p>
+                     </div>
                   </div>
                </div>
             </div>
          </div>
+         
       </div>
       {# 
       <div id="money-raised-accross-elections" class="entity__figure row">
```
Comment (johnnyporkchops) [inline fec/data/templates/partials/house-senate-overview/election-data.jinja:134]: Uncomment the anchor link to this ID in `sidebar-nav.jinja`

File: fec/fec/static/js/pages/elections-overview.js [hunk 3]
```diff
@@ -283,4 +287,83 @@ AcrossTime.prototype.loadData = function(query) {
...
+    },
+    columns: column_definitions,
+    order: [[0, 'asc']],
+    useFilters: true,
+    useExport: false,
+    lengthMenu: [10, 30, 50]
+  });
+  return $table;
+}
+
```
Comment (johnnyporkchops) [inline fec/fec/static/js/pages/elections-overview.js:351]: Nice job figuring out how to combine the two scripts!

File: fec/fec/static/js/pages/elections-overview.js [hunk 4]
```diff
@@ -283,4 +287,83 @@ AcrossTime.prototype.loadData = function(query) {
...
+$(function() {
+  new AcrossTime(); // setup across time display
+
```
Comment (johnnyporkchops) [inline fec/fec/static/js/pages/elections-overview.js:354]: Nice job figuring out how to combine the two scripts!

**Comment by patphongs** [comment]
Thanks @johnnyporkchops and @JonellaCulmer for all of the reviews. This is ready for you to take a look again.

File: fec/data/templates/partials/house-senate-overview/election-data.jinja [hunk 5]
```diff
@@ -152,30 +153,42 @@
...
       <div id="totals-for-all-elections" class="entity__figure row">
          <h2 class="heading__left">Totals for all {{ office | title }} elections</h2>
-         <p class="t-sans t-bold t-low-height">Compare the money raised, spent, cash on hand and debts by {{ office | title }} election</p>
-         {{ select.two_year_select(cycles, cycle, id="cycle-4") }}
-         <div class="content__section--ruled-responsive t-serif">
-            <div class="heading--with-action t-small u-negative--top--margin t-serif">
-               <i class="data-disclaimer">Some kind of disclaimer text that clarifies something high-level that users need to know right away to put the data in context.</i>
-               <div class="heading__right">
-                  <a class="button button--alt js-ga-event" data-a11y-dialog-show="totals-for-all-elections-modal" data-ga-event="Totals for all elections methodology modal clicked" aria-controls="totals-for-all-elections-modal">Methodology</a>
+         <p class="t-sans t-bold t-low-height">Compares the money raised, spent, cash on hand and debts by {{ office | title }} election</p>
```
Comment (johnnyporkchops) [inline fec/data/templates/partials/house-senate-overview/election-data.jinja:159]: this should not be bold  . We say "comparing"  in  the  across  time  feature

**Comment by patphongs** [comment]
> @JonellaCulmer We say "comparing" on across time feature. We should change that to match too, right?

Thanks @johnnyporkchops, as agreed upon by the content team, we're changing this to be "comparing" throughout the house and senate overview page

**Review by johnnyporkchops — APPROVED** [review]

**Pull request merged**